sdks/go/pkg/beam/core/runtime/exec/sideinput.go

```diff
@@ -33,6 +33,7 @@ const iterableSideInputKey = ""
 // encapsulates StreamID and coding as needed.
 type SideInputAdapter interface {
 	NewIterable(ctx context.Context, reader StateReader, w typex.Window) (ReStream, error)
+	GetIDs() (StreamID, string)
```

Comment:
TBH it still feels like the the Adapter is the right place for the cache to be looked up, since it's dealing with actual elements and similar, and produces the ReStream instances. It's already getting a StateReader that it can pull the cache from too (depends on what happens to the cache of course, it might be best if it's simply hidden within the reader entirely...).

Specifially, any time you need to add methods to access data elsewhere, the question is whether that has to be the case? What does the solution look like when you're not able/allowed to do that?

Reply:
That's a solution if we wanted to move the cache checking logic here, although it seems like it would still need a new function in the interface to handle it (changing the NewIterable signature to return a ReStream *or* a ReusableInput feels messy.)

Reply:
Changing to putting cache accesses in the Adapter cleaned up a lot of the null checks. Dropped GetIDs for a first run at cache-involved functions